packages/@aws-cdk/aws-cloudfront/test/integ.cloudfront-alias-target.ts

```diff
@@ -0,0 +1,30 @@
+import route53 = require('@aws-cdk/aws-route53');
+import s3 = require('@aws-cdk/aws-s3');
+import cdk = require('@aws-cdk/cdk');
+import cloudfront = require('../lib');
+
+const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'aws-cdk-cloudfront');
+
+const zone = new route53.PublicHostedZone(stack, 'HostedZone', { zoneName: 'test.public' });
+
+const sourceBucket = new s3.Bucket(stack, 'Bucket');
+
+const distribution = new cloudfront.CloudFrontWebDistribution(stack, 'MyDistribution', {
+  originConfigs: [
+    {
+      s3OriginSource: {
+        s3BucketSource: sourceBucket
+      },
+      behaviors : [ {isDefaultBehavior: true}]
+    }
+  ]
+ });
+
+new route53.AliasRecord(zone, 'Alias', {
```

Comment:
Ew, do records have to be constructed under the zone that they need to be created in?

That's not according to how we design most of the construct library. But the R53 library is quite old, so it makes sense that it's not up-to-date with the latest guidelines.

Not your problem.